### PR TITLE
Swapped Jesque for Redisq

### PR DESCRIPTION
**Pull request by pluraliseseverythings**

### CODE REVIEW

File: grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java [hunk 1]
```diff
@@ -272,126 +271,27 @@ void sendQueriesToLoader(Collection<Query> queries){
                 .withWaitStrategy(WaitStrategies.fixedWait(1, TimeUnit.SECONDS))
                 .build();
 
-        try {
```
Comment (pluraliseseverythings) [inline grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java]: I deleted a lot of logic for making sure the number of batches in flight is X. I don't see why we need to check that explicitly. We can just submit a job in a fixed size thread pool that enforces that limit

File: grakn-engine/src/main/java/ai/grakn/engine/controller/TasksController.java [hunk 2]
```diff
@@ -422,18 +420,7 @@ private Json serialiseStateSubset(TaskState state) {
         return Json.object()
                 .set(ID, state.getId().getValue())
                 .set(STATUS, state.status().name())
-                .set(CREATOR, state.creator())
```
Comment (pluraliseseverythings) [inline grakn-engine/src/main/java/ai/grakn/engine/controller/TasksController.java]: I deleted a bunch of stuff that wasn't used. If we want recurring and stoppable tasks, we change the redisq library

**Comment by pluraliseseverythings** [comment]
rtg

**Comment by pluraliseseverythings** [comment]
!rtg

File: docs/pages/documentation/6-developing-with-java/loader-api.md [hunk 3]
```diff
@@ -176,6 +178,15 @@ loader.setRetryPolicy(true);
 loader.setRetryPolicy(false);
 ```
 
+
+## Close
+
+The loader can be closed as follows
+
+```java
+loader.close();
```
Comment (fppt) [inline docs/pages/documentation/6-developing-with-java/loader-api.md]: Brilliant we forgot that.

File: grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java [hunk 4]
```diff
@@ -245,25 +267,20 @@ public void waitToFinish(){
      * @param queries Queries to be inserted
      */
     void sendQueriesToLoader(Collection<Query> queries){
-        try {
-            blocker.acquire();
```
Comment (fppt) [inline grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java]: YEEESSSSS!!!!! . . .  What the hell did this actually do????

File: grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java (hunk 1 above)
Comment (fppt) [reply grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java]: And thank god for that. I was looking at all thise on friday and thinking to myself that it seemed far too complex for something sooooo simple

File: grakn-core/src/main/java/ai/grakn/engine/TaskStatus.java [hunk 5]
```diff
@@ -52,4 +54,38 @@
...
+        switch (this) {
+            case FAILED:
+                return State.FAILED;
+            case STOPPED:
+                return State.PROCESSING;
+            case COMPLETED:
+                return State.DONE;
+            case RUNNING:
+                return State.PROCESSING;
+            case SCHEDULED:
```
Comment (fppt) [inline grakn-core/src/main/java/ai/grakn/engine/TaskStatus.java]: could probably just compress `SCHEDULED` and `CREATED` into the default case

File: grakn-core/src/main/java/ai/grakn/engine/TaskStatus.java [hunk 6]
```diff
@@ -52,4 +54,38 @@
...
+
+    public static TaskStatus fromState(State state) {
+        switch (state) {
+            case FAILED:
+                return FAILED;
+            case PROCESSING:
+                return RUNNING;
+            case DONE:
+                return COMPLETED;
+            case NEW:
```
Comment (fppt) [inline grakn-core/src/main/java/ai/grakn/engine/TaskStatus.java]: Same thing for `NEW`.

File: grakn-kb/src/main/java/ai/grakn/kb/internal/cache/GlobalCache.java [hunk 7]
```diff
@@ -33,33 +33,29 @@
...
- *     Caches Knowledge Base or Session specific data which is shared across transactions:
- *     <ol>
- *         <li>Schema Cache - All the types which make up the schema. This cache expires</li>
- *         <li>
- *             Label Cache - All the labels which make up the schema. This can never expire and is needed in order
- *             to perform fast lookups. Essentially it is used for mapping labels to ids.
- *         </li>
- *     <ol/>
- * </p>
+ * <p> Caches Knowledge Base or Session specific data which is shared across transactions: <ol>
```
Comment (fppt) [inline grakn-kb/src/main/java/ai/grakn/kb/internal/cache/GlobalCache.java]: Errrr formatting went wonky here.

File: pom.xml [hunk 8]
```diff
@@ -196,6 +195,11 @@
 
     <dependencies>
         <dependency>
+            <groupId>ai.grakn</groupId>
+            <artifactId>redisq</artifactId>
+            <version>0.0.2-SNAPSHOT</version>
```
Comment (fppt) [inline pom.xml]: Need a property for this version

File: grakn-test/test-integration/src/test/java/ai/grakn/test/migration/json/JsonMigratorMainTest.java [hunk 9]
```diff
@@ -110,6 +109,7 @@ public void jsonMigratorCalledInvalidTemplateFile_ErrorIsPrintedToSystemErr(){
     }
 
     @Test
+    @Ignore
```
Comment (fppt) [inline grakn-test/test-integration/src/test/java/ai/grakn/test/migration/json/JsonMigratorMainTest.java]: TP task to undo this ignore?

File: grakn-test-tools/src/main/java/ai/grakn/util/EmbeddedRedis.java [hunk 10]
```diff
@@ -76,19 +74,36 @@ public static void start(int port){
...
             redisServer.stop();
             LOG.info("Redis stopped.");
-        } else {
-            LOG.warn("Called stop while {} redis instances are running", REDIS_COUNTER);
+        } catch (Exception e) {
+            LOG.warn("Failure while stopping redis", e);
+        }
+    }
+
+    public static void forceStart(int redisPort) {
```
Comment (fppt) [inline grakn-test-tools/src/main/java/ai/grakn/util/EmbeddedRedis.java]: The fact that this method exists makes me think we should REALLY try to mock out redis properly.

File: grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java (hunk 4 above)
Comment (pluraliseseverythings) [reply grakn-client/src/main/java/ai/grakn/client/BatchMutatorClient.java]: I can't remember exactly but there was a lot of extra logic to make sure we had at most N requests running in parallel, which can be done with a threadpool served by a blocking queue and no extra synchronisation

File: grakn-test/test-integration/src/test/java/ai/grakn/test/migration/json/JsonMigratorMainTest.java (hunk 9 above)
Comment (pluraliseseverythings) [reply grakn-test/test-integration/src/test/java/ai/grakn/test/migration/json/JsonMigratorMainTest.java]: Well spotted, let me see if I can put it back

**Comment by pluraliseseverythings** [comment]
!rtg

**Comment by pluraliseseverythings** [comment]
!rtg

**Comment by burukuru** [comment]
!rtg

**Comment by pluraliseseverythings** [comment]
!rtg

**Comment by pluraliseseverythings** [comment]
!rtg

**Comment by pluraliseseverythings** [comment]
!rtg

**Pull request merged**